File: src/main/cpp/commands/DefaultTankDriveCommand.cpp
#include "commands/DefaultTankDriveCommand.h"

DefaultTankDriveCommand::DefaultTankDriveCommand(Drivetrain *subsystem, std::function<double()> left, std::function<double()> right)
    : m_drivetrain{subsystem}, m_left{std::move(left)}, m_right{std::move(right)}
{
  AddRequirements({subsystem});
}

void DefaultTankDriveCommand::Execute()
{
  m_drivetrain->TankDrive(m_left(), m_right());
}

File: src/main/java/frc/robot/commands/DefaultTankDriveCommand.java
package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.subsystems.drivetrain.Drivetrain;

public class DefaultTankDriveCommand extends CommandBase {
 private final Drivetrain m_drivetrain;
  private final double m_leftSpeed;
  private final double m_rightSpeed;
  
  public DefaultTankDriveCommand(Drivetrain drivetrain, double leftSpeed, double rightSpeed) {
    m_drivetrain = drivetrain;
    m_leftSpeed = leftSpeed;
    m_rightSpeed = rightSpeed;
    addRequirements(drivetrain);
  }

  @Override
  public void execute() {
    m_drivetrain.tankDrive(m_leftSpeed, m_rightSpeed);
  }
}
